src/bluesearch/database/article.py

```diff
@@ -0,0 +1,246 @@
+# Blue Brain Search is a text mining toolbox focused on scientific use cases.
+#
+# Copyright (C) 2020  Blue Brain Project, EPFL.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+"""Abstraction of scientific article data and related tools."""
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Generator, List, Sequence, Tuple, Type, TypeVar
+
+# This is for annotating the return value of the Article.parse class method, see
+# https://github.com/python/typing/issues/254#issuecomment-661803922
+_T = TypeVar("_T", bound="Article")
+
+
+class ArticleParser(ABC):
+    """An abstract base class for article parsers."""
+
+    @abstractmethod
+    def get_title(self) -> str:
```

Comment:
For instance
```suggestion
    @property
    @abstractmethod
    def title(self) -> str:
```

so that
- `print(article.title)` will work correctly
- `article.title = "Another article"` will fail with `AttributeError: can't set attribute`.

Reply:
Related changes: f68fbec